content/docs/guides/benchmarking-latency.md

```diff
@@ -0,0 +1,107 @@
+
+## Connection types and their impact
+
+Neon supports two connection protocols with distinctly different performance profiles. Understanding these differences is crucial for accurate benchmarking. For a comprehensive comparison, see [Choosing Connection Types](/docs/connect/choose-connection):
+
+### HTTP connections
+
+- **Performance profile**: Optimized for single-shot queries with minimal connection overhead
+- **Use cases**: Ideal for serverless functions that execute a single query per invocation
+- **Limitations**: Doesn't support sessions, interactive transactions, NOTIFY, or COPY protocol
+- **When to benchmark**: Use for measuring performance of stateless, individual query operations
+- **Optimization technique**: Connection caching can further reduce latency for HTTP connections
+
+### WebSocket connections
+
+- **Performance profile**: Higher initial connection overhead but significantly faster for subsequent queries
+- **Use cases**: Optimal for applications that execute multiple queries over a maintained connection
+- **Features**: Supports full PostgreSQL functionality including sessions, transactions, and all Postgres protocols
+- **When to benchmark**: Measure both connection establishment time AND subsequent query execution separately
+- **Initialization**: Requires multiple round-trips between client and server to establish
+
+For implementation details on both connection methods, refer to the [Serverless Driver Documentation](/docs/serverless/serverless-driver).
+
+**Benchmarking considerations**: When comparing HTTP vs WebSocket connections, you'll typically observe a bi-modal distribution in query latencies. HTTP connections generally show more consistent performance for individual queries, while WebSocket connections show higher initial latency followed by very low latency for subsequent operations. The runtime environment (Edge vs traditional Serverless) can also impact connection performance characteristics.
+
+**Testing approach**: For WebSockets, establish the connection first, then measure query execution time separately. This reflects real-world usage where connections are reused rather than created for each query. For HTTP, measure individual query execution time including any per-query connection overhead.
+
+## Real-world usage pattern simulation
+
+To get meaningful results, design your benchmarks to simulate how your application actually interacts with Neon:
+
+- **Use persistent connections**: For web servers or long-running applications, initialize the database connection before measuring query timings, then run a series of queries on this persistent connection. Consider [Connection Pooling](/docs/connect/connection-pooling) for optimal performance.
```

Comment:
Maybe make connection pooling its own list item.

Reply:
Thank you, connection pooling info will be expanded.